reader_concurrency_semaphore.cc

```diff
@@ -864,35 +864,58 @@ future<> reader_concurrency_semaphore::enqueue_waiter(reader_permit permit, read
 }
 
 void reader_concurrency_semaphore::evict_readers_in_background() {
+    if (_evicting) {
+        return;
+    }
+    _evicting = true;
     // Evict inactive readers in the background while wait list isn't empty
     // This is safe since stop() closes _gate;
     (void)with_gate(_close_readers_gate, [this] {
         return do_until([this] { return _wait_list.empty() || _inactive_reads.empty(); }, [this] {
             return detach_inactive_reader(_inactive_reads.front(), evict_reason::permit).close();
         });
-    });
- }
+    }).finally([this] { _evicting = false; });
+}
+
+reader_concurrency_semaphore::can_admit
+reader_concurrency_semaphore::can_admit_read(reader_permit permit, wait_list_should_be_empty wait_list_empty) const noexcept {
+    if (_wait_list.empty() != bool(wait_list_empty)) {
```

Comment:
I guess that the `wait_list_should_be_empty::no` case means "don't care" rather than strictly not empty,
i.e. this condition should be something like `if (wait_list_empty && !_wait_list.empty())` (should be no need for the explicit cast to bool with the boolean operator).
That said, maybe a better name for it would be `require_empty_wait_list`.

Reply:
Done in v2.